crates/full-node/sov-db/src/test_utils.rs

```diff
@@ -290,3 +290,41 @@ fn is_version_selected_for_key(
     // This is an approximation, but much more efficient than the exact method
     random_value < selection_probability
 }
+
+use strum::{Display, EnumString};
+
+/// This environment variable sets the crash location for rollup and is used only in tests.
+pub const CRASH_ENV_NAME: &str = "SOV_CRASH_ON_COMMIT";
+
+/// The crash location.
+#[derive(Debug, Clone, Display, EnumString, Eq, PartialEq)]
+pub enum CrashLocation {
+    /// Rollup crashes before saving `the CommittingKernelNomt` flag.
+    BeforeSavingKernelNomt,
+    /// Rollup crashes before committing the kernel.
+    BeforeCommittingKernelNomt,
+    /// Rollup crashes before saving the `CommittingUserNomt` flag.
+    BeforeSavingUserlNomt,
+    /// Rollup crashes before committing the user nomt.
+    BeforeCommittingUserNomt,
```

Comment:
I think it is very important to add failures after commit is done, but before flag is written.